Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py

```diff
@@ -0,0 +1,152 @@
+from datetime import datetime
+from .parsers import DateTimeParser, DateTimeParseResult
+from .constants import Constants
+from recognizers_text import ExtractResult, ParseResult, QueryProcessor
+from .utilities import DateTimeResolutionResult, MatchingUtil
+from recognizers_date_time import DateTimeZoneExtractor
+from .utilities import Token
+from .utilities import DateTimeOptionsConfiguration
+from recognizers_text.matcher.string_matcher import StringMatcher
+from abc import abstractmethod
+from typing import List, Optional, Pattern, Match, Dict
+import regex
+from recognizers_text.matcher.match_result import MatchResult
+
+
+class BaseTimeZoneParser(DateTimeParser):
+    @property
+    def parser_name(self) -> str:
+        return Constants.SYS_DATETIME_TIMEZONE
+
+    def compute_minutes(self, utc_offset: str) -> int:
+        pass
+
+    def convert_offset_in_mins_to_offset_string(self, offset_mins: int) -> str:
+        pass
+
+    def convert_mins_to_regular_format(self, offset_mins: int) -> str:
+        pass
+
+    def normalize_text(self, text: str) -> str:
+        pass
+
+    def parse(self, extract_result: ExtractResult, ref_date: datetime = None) -> ParseResult:
+        pass
+
+    def filter_results(self, query: str, candidate_results: List[DateTimeParseResult]) -> List[DateTimeParseResult]:
+        pass
+
+    def parse(self, extract_result: ExtractResult, ref_date: datetime) -> DateTimeParseResult:
+        pass
+
+    def get_datetime_resolution_result(self, offset_mins: int, text: str) -> DateTimeResolutionResult:
+        pass
+
+
+class TimeZoneExtractorConfiguration(DateTimeOptionsConfiguration):
+    @property
+    @abstractmethod
+    def direct_utc_regex(self) -> regex:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def location_time_suffix_regex(self) -> regex:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def location_matcher(self) -> StringMatcher:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def time_zone_matcher(self) -> StringMatcher:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def ambiguous_time_zone_list(self) -> List[str]:
+        raise NotImplementedError
+
+
+class BaseTimeZoneExtractor(DateTimeZoneExtractor):
+    @property
+    def extractor_type_name(self) -> str:
+        return Constants.SYS_DATETIME_TIME
+
+    def __init__(self, config: TimeZoneExtractorConfiguration):
+        self.config = config
+
+    def extract(self, source: str, reference: datetime = None) -> List[ExtractResult]:
```

Comment:
You are missing an _extract()_ method in this class. Can you add it? Also, to implement it, use default values.